src/cookbooks/clients_last_seen_bits.md

```diff
@@ -0,0 +1,136 @@
+   
+  └──────────────────────────────────────────────────────────────────────────────────┘
+  MAU                                                             └──────────────────┘
+                                                                   WAU             └─┘
+                                                                                   DAU
+```
+
+In this picture, we've annotated the windows for DAU (day 0 only), 
+WAU (days 0 through -6) and MAU (days 0 through -27). This particular client
+won't count toward DAU for 2020-01-28, but the client does count towards both
+WAU and MAU.
+
+You may notice that the day 13 bit doesn't end up affecting these metrics.
+For DAU, WAU, and MAU, all that matters is the rightmost bit that's set,
+telling us about the most recent activity seen for that user. We provide
+a special function for that use case:
+
+
+```
+SELECT udf.bits28_days_since_seen(8256)
+-- 6
+```
+
+Indeed, this is so commonly used that we build this function into user-facing
+views, so that instead of referencing `days_seen_bits` with a UDF, you can
+instead reference a field called `days_since_seen`. Counting MAU, WAU, and
+DAU generally looks like:
+
+```
+SELECT
+  COUNT(days_since_seen < 28) AS mau,
+  COUNT(days_since_seen <  7) AS wau,
+  COUNT(days_since_seen <  1) AS dau
+FROM
+  telemetry.clients_last_seen
+```
+
+Note that this particular client is about to fall outside the WAU window.
+If the client doesn't send a main ping on 2020-01-29, the new `days_seen_bits`
+pattern will look like:
+
+```
+    0  0  0  0  0  0  0  0  0  0  0  0  0  1  0  0  0  0  0  0  1  0  0  0  0  0  0  0
+    ──────────────────────────────────────────────────────────────────────────────────
+    │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │
+    │ 26  │ 24  │ 22  │ 20  │ 18  │ 16  │ 14  │ 12  │ 10  │  8  │  6  │  4  │  2  │  0
+   27    25    23    21    19    17    15    13    11     9     7     5     3     1
+   
+  └──────────────────────────────────────────────────────────────────────────────────┘
+  MAU                                                             └──────────────────┘
+                                                                   WAU             └─┘
+                                                                                   DAU
+```
+
+The `days_seen_bits` value is now `16512` and the user is indeed outside the WAU window:
+
+```
+SELECT udf.bits28_days_since_seen(16512)
+-- 7
+```
+
+## Retention: Forward-looking windows
+
+For retention calculations, we user forward-looking windows. This means that
+when we report a retention value for 2020-01-01, we're talking about what
+portion of clients active on 2020-01-01 are still active some number of days
+later.
+
+When we were talking about backward-looking windows, our anchor date or "day 0"
+was always the `submission_date`. When we define forward-looking windows, however,
```

Comment:
What does the `submission_date` actually means here? I assume you mean `submission_date` of clients_last_seen?

Reply:
Yes, I'll try to make this more clear.

Reply:
One could argue that we should not use the name `submission_date` at all in these tables, since this is far enough from raw pings that there is no reasonable definition of when anything was "submitted".  I haven't thought deeply about this though, so feel free to disagree :)

Reply:
I prefer more specificity.  If these were all derived from submission dates that makes sense to me. If they are all different kinds of dates that were processed downstream to fit into a model of the truth we can use some other word.  I have a strong aversion to just using 'date' because it is a SQL function and I have to start putting quotes on it all the time.